homeassistant/components/tellstick/sensor.py

```diff
@@ -35,6 +36,8 @@
                     {
                         vol.Required(CONF_ID): cv.positive_int,
                         vol.Required(CONF_NAME): cv.string,
+                        vol.Optional(CONF_PROTOCOL, default=""): cv.string,
```

Comment:
Don't default to empty string. Fetch the option with `dict.get` instead to default to `None`.

Reply:
That was my first intention, but then I read
"Default values for optional configuration keys need to be valid values. **Don't use a default which is None like** vol.Optional(CONF_SOMETHING, default=None): cv.string, set the default to default='' if required."
https://developers.home-assistant.io/docs/en/development_validation.html

Reply:
Yes. Remove the default here. Then use `dict.get` below.